pages/data-providers/[data-provider-id]/settings.jsx

```diff
@@ -1,84 +1,103 @@
-import React, { useReducer, useCallback, useState } from 'react'
+import React, { useState, useEffect } from 'react'
 import { classNames } from '@oacore/design/lib/utils'
 
 import styles from './settings.css'
 
 import { Card, TextField, Button } from 'design'
 import { withGlobalStore } from 'store'
 
-const Settings = ({ store, className, ...restProps }) => {
-  const [isSaving, setIsSaving] = useState(false)
-  const [hasFormChanged, setHasFormChanged] = useState(false)
-
-  const reducer = (state, { name, value }) => {
-    setHasFormChanged(true)
-    return {
-      ...state,
-      [name]: value,
-    }
+const SettignsGroup = ({
+  children,
+  className,
+  tag,
+  onSubmit,
+  ...formProps
+}) => {
+  const [isChanged, setChanged] = useState(false)
+  const [isSubmited, setSubmitted] = useState(false)
+
+  const handleChange = () => setChanged(true)
```

Comment:
Would be nice to hide/disable the button when changes are returned to default.

Reply:
It would but it's not necessary. It's quite small thing and the page it not really popular.